### PR TITLE
DynamicDependencies: fix PCWSTR[] access

### PR DESCRIPTION
**Pull request by DrusTheAxe**
Also fixed cert access in DevCheck.ps1 causing false errors

### CODE REVIEW

**Comment by jonwis** [comment]
Is there a test that was failing, and now won't be?

What issue # are you fixing?

File: dev/DynamicDependency/PackageDependency.cpp [hunk 1]
```diff
@@ -42,9 +42,9 @@ std::vector<std::wstring> MddCore::PackageDependency::FindPackagesByFamily() con
     THROW_IF_WIN32_ERROR(FindPackagesByPackageFamily(m_packageFamilyName.c_str(), PACKAGE_FILTER_HEAD | PACKAGE_FILTER_DIRECT, &count, packageFullNames.get(), &bufferLength, buffer.get(), nullptr));
 
     std::vector<std::wstring> packageFullNamesList;
-    auto packageFullName{ *packageFullNames.get() };
-    for (UINT32 index=0; index < count; ++index, ++packageFullName)
+    for (UINT32 index=0; index < count; ++index)
     {
+        const auto packageFullName{ packageFullNames[index] };
         packageFullNamesList.push_back(std::wstring(packageFullName));
```
Comment (jonwis) [inline dev/DynamicDependency/PackageDependency.cpp:48]:
```suggestion
        packageFullNamesList.emplace_back(packageFullNames[index]);
```

File: tools/DevCheck.ps1 [hunk 2]
```diff
@@ -255,14 +255,15 @@ function Test-DevTestCert
     $pfx = Get-PfxCertificate -FilePath $path
     $thumbprint = $pfx.Thumbprint
 
-    $cert = Get-ChildItem -Path "cert:\LocalMachine\TrustedPeople\$thumbprint"
```
Comment (jonwis) [inline tools/DevCheck.ps1:258]: Another pattern for this is:

```ps1
$cert = Get-ChildItem -Path "cert:\LocalMachine\TrustedPeople\$thumbprint" -ErrorAction SilentlyContinue;
if ($cert)
{
    // use cert
}
else
{
    // complain
}
```

Avoids two tests, lots of red, and a "not foo"

**Review by huichen123 — APPROVED** [review]

**Review by jonwis — APPROVED** [review]
Minor comments

**Comment by DrusTheAxe** [comment]
> Is there a test that was failing, and now won't be?
Wasn't noticed due to a test gap. Closing that gap is tricky so won't happen this week (I'm on VACA and traveling, just happened to see the issue and have a few...). But @AdamBraden has an easy manual repro,

> What issue # are you fixing?
@AdamBraden  just created one. Microsoft-internal task [33487794](https://task.ms/33487794)

File: tools/DevCheck.ps1 (hunk 2 above)
Comment (DrusTheAxe) [reply tools/DevCheck.ps1:258]: -ErrorAction is cute. Thanks for the tip. Will look into it [when next in here](https://github.com/microsoft/ProjectReunion/issues?q=is%3Aissue+is%3Aopen+devcheck)

**Review by BenJKuhn — APPROVED** [review]

**Pull request merged**